### PR TITLE
Add ZSink#sum and ZSink#count (#2736 #2737)

### PR DESCRIPTION
**Pull request by kolemannix**

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/ZSink.scala [hunk 1]
```diff
@@ -1302,6 +1302,19 @@ object ZSink extends ZSinkPlatformSpecificConstructors with Serializable {
         p(a).map(if (_) ((a :: as, true), Chunk.empty) else ((as, false), Chunk.single(a)))
     }.map(_._1.reverse)
 
+  /**
+   * Creates a sink which emits the number of elements processed
+   */
+  def count[A]: ZSink[Any, Nothing, Nothing, A, Long] = {
+    new SinkPure[Nothing, Nothing, A, Long] {
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/ZSink.scala]: To cut down on repetition, let's implement this with `Sink.foldLeft`?

File: streams/shared/src/main/scala/zio/stream/ZSink.scala [hunk 2]
```diff
@@ -1835,6 +1848,20 @@ object ZSink extends ZSinkPlatformSpecificConstructors with Serializable {
       def cont(state: State)           = false
     }
 
+  /**
+   * Creates a sink which sums elements, provided they are Numeric
+   */
+  def sum[A](implicit ev: Numeric[A]): ZSink[Any, Nothing, Nothing, A, A] = {
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/ZSink.scala]: To cut down on repetition, let's implement this with `Sink.foldLeft`?

File: streams/shared/src/main/scala/zio/stream/ZSink.scala (hunk 1 above)
Comment (kolemannix) [reply streams/shared/src/main/scala/zio/stream/ZSink.scala]: 👍

File: streams/shared/src/main/scala/zio/stream/ZSink.scala (hunk 2 above)
Comment (kolemannix) [reply streams/shared/src/main/scala/zio/stream/ZSink.scala]: 👍

**Comment by iravid** [comment]
Ah, looks like tests are failing on Dotty. Could you check that?

**Pull request merged**